### DISCUSSION TITLE
Exposición  «Firmware: De BIOS a UEFI»: preguntas y respuestas

### DISCUSSION THREAD

**Discussion by gwolf** [unamfi/sistop-2022-1#162] [Q&A]
Abro este foro como un espacio de discusión para la exposición que preparó el alumno Santos Albañil Morteo, «Firmware: De BIOS a UEFI». El video está disponible en:

https://www.youtube.com/watch?v=ejz7lNN5XG4

Pueden consultar también el documento de texto en que los alumnos presentan este mismo tema:

https://github.com/unamfi/sistop-2022-1/blob/main/exposiciones/AlbañilSantos/Firmware_SantosAM.pdf

Además de participar aquí, les pido que llenen la pequña encuesta que está disponible en:

https://www.questionpro.com/t/ARiwuZp7oS

**Comment by JesusESD** [comment]
Saludos compañero Santos! Me pareció muy completa tu presentación; en general me gustó mucho. De asignaturas anteriores, recuerdo que también se empleaban las siglas "EFI" para referirse a un concepto en este tema. ¿Existe alguna diferencia entre EFI y UEFI? ¿O hacen referencia a lo mismo? Agradeceré mucho tu respuesta :)

**Reply by Santos915** [reply]
EFI es su primer nombre, originalmente era desarrollado por Intel y se llamaba EFI(Extensible Firmware Interface), cuando se libero paso a ser UEFI(Unified Extensible Firmware Interface)...

**Comment by albertogfm** [comment]
Buena presentación compañero, mi duda es: ¿Los servidores web alojados en las grandes empresas usan UEFI para su hardware o existe alternativa para esos casos?

**Reply by Santos915** [reply]
De echo fueron los primeros en usarlo, sus requisitos son mas elevados en muchos aspectos, recordemos que uno de los aspectos es la manipulacion de disco duros de gran tamaño, que en BIOS esta limitado (entre otras cosas).

**Reply by Santos915** [reply]
¡UEFI es la alternativa!

**Reply by gwolf** [reply]
Y tal vez no nos damos tanto cuenta de esto en el mundo x86; en ARM, el inicio temprano era siempre un soberano dolor de cabeza. Conforme los equipos ARM comienzan a adoptar el estándar UEFI, se va volviendo muchísimo más sencillo el soporte para sistemas operativos en ellos.

**Comment by jorgelmp** [comment]
Hola, compañero. Tu presentación me pareció muy interesente. Me queda una pregunta, ¿quién se encarga de implementar UEFI? ¿cada fabricante de hardaware? ¿en ese caso UEFI es una especificación que deben seguir? ¿o hay algo que estoy entendiendo completamente mal?

**Reply by Santos915** [reply]
Exactamente. UEFI es una especificacion, hay varios provedores de firmware que lo producen (realmente no muchos), se puede trabajar sobre codigo ya echo, como EDK II (en teoria al contar con la especificacion podemos desarrollarlo completo desde 0), luego le añaden una capa personalizada para cada marca o modelo de producto(la parte que vemos al configurar el ""bios"")

**Comment by Inakyord** [comment]
Hola, mi pregunta creo que es un poco especulativa, pero supongo que al haber investigado el tema puedes responderla mejor. ¿Hoy en día queda algún beneficio de utilizar BIOS para un equipo? o ¿Eventualmente cuando se cambien las máquinas que usan BIOS por unas modernas éste va a desaparecer por completo?

**Reply by Santos915** [reply]
BIOS ya fue completamente sustituido por UEFI, si compras un producto nuevo ya no encotraras BIOS, solo UEFI. BIOS continua presente en equipo relativamente viejo, o en equipos que manejen SO no compatibles con UEFI(aunque UEFI tiene un modulo de compatibilidad que emula BIOS, para estos casos).
